### ISSUE TITLE
Unable to use CLI --list option

### ISSUE THREAD

**Issue by pboling** [skryukov/rubocop-gradual#21]
I am using the rubocop-gradual shim, so it runs instead of the standard rubocop command normally.

When I run:
```
NO_GRADUAL=1 bundle exec rubocop -a --list
```
it only prints the list of files that would be processed.

If I instead run:
```
bundle exec rubocop -a --list
```

It runs rubocop gradual ignoring the `--list` option entirely.

Is this intentional?

**Comment by pboling** [comment]
I'm trying to understand why my Gemfile is linted when I run standard RuboCop, but not when I run RuboCop gradual.

In fact when I use `NO_GRADUAL=1` I see:
```
Inspecting 436 files
```

But when I run rubocop-gradual I see:

```
Inspecting 81 file(s) for autocorrection...
.................................................................................
Fixed 0 file(s).
....................................................................................................................................................................................................................................................................................................................................................................................................................................................
Found 355 files with 13150 issue(s).
```

It seems that rubocop-gradual is unable to see ~ 80 of my project's files that rubocop would process if it were not for rubocop-gradual being loaded.

**Comment by pboling** [comment]
I'm guessing that this is relevant somehow (and also not entirely fixed):
https://github.com/rubocop/rubocop/issues/5251

**Comment by skryukov** [comment]
Hey, sorry for the late response! I made `rubocop-gradual -a` to run autocorrection only for all new and changed files, this makes sense for my application – fixing files that I touched in a current PR. So, my guess is Gemfile wasn't updated when you ran the command 😅

Regarding `--lint` – it seems to be useful to implement it in rubocop-gradual 🤔

**Comment by pboling** [comment]
It's not `--lint`, but `--list`. :)

Also, the issue isn't related to file changes.  The huge difference between running vanilla `rubocop` and `rubocop_gradual` is with zero file changes in the app at all, and a prior clean run (`force_update` if needed) of `rubocop_gradual`.

Simply from changing the command I run from rubocop to rubocop_gradual, the files processed, and the rules used during that processing, are totally different.

**Comment by skryukov** [comment]
> Simply from changing the command I run from rubocop to rubocop_gradual, the files processed, and the rules used during that processing, are totally different.

Yup, `rubocop-gradual` changes `-a` and `-A` behavior in require mode, that's kinda by design 😅

The change in rules is strange, but I didn't find anything related to that yet 😫

**Comment by pboling** [comment]
> Yup, rubocop-gradual changes -a and -A behavior in require mode, that's kinda by design 😅

But it should only change which files are modified by RuboCop gradual, not which files are "found", right?

It has to be able to, at minimum, find every file that RuboCop vanilla can find, so that it can generate a hash of the file to track in the .lock.  I guess that's my core issue.  The .lock has nearly 100 fewer files in it than vanilla rubocop is evaluating.

**Comment by pboling** [comment]
@skryukov The number of files inspected is still changing wildly in unexpected ways.  I'll update this with `--debug` info as well.
